src/keyshare_export.rs

```diff
@@ -0,0 +1,162 @@
+//! Module to export and import key shares, with encryption in transit.
+
+use crate::{errors::Result, keygen::KeySharePrivate};
+use std::{marker::PhantomData, ops::Deref};
+
+mod pke;
+pub use pke::Pke;
+
+mod sodium;
+pub use sodium::SodiumPke;
+
+/// Struct for handling the export and import of KeySharePrivate
+#[derive(Clone, Debug)]
+pub struct KeyShareEncrypted<T: Pke>(Vec<u8>, PhantomData<T>);
+
+impl<T: Pke> From<Vec<u8>> for KeyShareEncrypted<T> {
+    fn from(data: Vec<u8>) -> Self {
+        Self(data, PhantomData)
+    }
+}
+
+impl<T: Pke> From<KeyShareEncrypted<T>> for Vec<u8> {
```

Comment:
I think we should remove this `From` instance and simply provide a getter to the underlying data.

Reply:
Done.